src/uct/ib/ud/base/ud_iface.c

```diff
@@ -350,13 +260,67 @@ static void uct_ud_iface_timer(int timer_id, int events, void *arg)
     uct_ud_iface_async_progress(iface);
 }
 
+static const void *
+uct_ud_iface_conn_match_get_peer_address(const ucs_conn_match_elem_t *elem)
+{
+    uct_ud_ep_t *ep = ucs_container_of(elem, uct_ud_ep_t, conn_match);
+    return uct_ud_ep_get_peer_address(ep);
+}
+
+static ucs_conn_sn_t
+uct_ud_iface_conn_match_get_conn_sn(const ucs_conn_match_elem_t *elem)
+{
+    uct_ud_ep_t *ep = ucs_container_of(elem, uct_ud_ep_t, conn_match);
+    return ep->conn_sn;
+}
+
+static const char *
+uct_ud_iface_conn_match_peer_address_str(const ucs_conn_match_ctx_t *conn_match_ctx,
+                                         const void *address,
+                                         char *str, size_t max_size)
+{
+    uct_ud_iface_t *iface      = ucs_container_of(conn_match_ctx,
+                                                  uct_ud_iface_t,
+                                                  conn_match_ctx);
+    uct_ud_iface_ops_t *ud_ops = ucs_derived_of(iface->super.ops,
+                                                uct_ud_iface_ops_t);
+    return ud_ops->peer_address_str(iface, address, str, max_size);
+}
+
+static void
+uct_ud_iface_conn_match_purge_cb(ucs_conn_match_ctx_t *conn_match_ctx,
+                                 ucs_conn_match_elem_t *elem)
+{
+    uct_ud_iface_t *iface      = ucs_container_of(conn_match_ctx,
+                                                  uct_ud_iface_t,
+                                                  conn_match_ctx);
+    uct_ud_iface_ops_t *ud_ops = ucs_derived_of(iface->super.ops,
+                                                uct_ud_iface_ops_t);
+    uct_ud_ep_t *ep            = ucs_container_of(elem, uct_ud_ep_t,
+                                                  conn_match);
+
```

Comment:
maybe add macro to call IB or UD iface op function?

Reply:
@yosefe if we are going to re-use this helper macro in other IB and SM TLs, better consider for the next separate PR

Reply:
ok